Gems/GraphModel/Code/Source/Integration/GraphController.cpp

```diff
@@ -698,7 +698,10 @@ namespace GraphModelIntegration
         GraphModel::NodePtrList nodeList;
         for (auto nodeId : nodeIds)
         {
-            nodeList.push_back(m_elementMap.Find<GraphModel::Node>(nodeId));
+            if (GraphModel::NodePtr nodePtr = m_elementMap.Find<GraphModel::Node>(nodeId))
```

Comment:
This will fail the `GraphModelIntegrationTests, GetNodesFromGraphNodeIds` test because it tests both sending an invalid `NodeId` and a valid `NodeId`, but one that isn't in the scene and checks to verify a nullptr was returned, so that test will need to be updated if this is the intended new behavior.

Reply:
Some additional info:

From what I gathered when I was debugging this, the `nodeId` that the GraphModel gem is passing in was valid for the scene ... but not for an item that `GraphModel::Node` derives from. Basically, I was creating a custom graph editor, but when I selected the connection *between* nodes I received a crash because I assumed I would only receive valid nodes back from `GetSelectedNodes`:

```
GraphModel::NodePtrList result;
GraphModelIntegration::GraphControllerRequestBus::EventResult(
	result,
	this->GetActiveGraphCanvasGraphId(), &GraphModelIntegration::GraphControllerRequestBus::Events::GetSelectedNodes);
	
if (result.empty())
	{
		m_propertyEditorCard->setEnabled(false);
		return;
	}

const GraphModel::SlotPtr dialogueDataSlotPtr = result.front()->GetSlot(CommonSlotNames::DIALOGUEDATA); 
//crash from front() returning nullptr.
```
![image](https://user-images.githubusercontent.com/47983418/140435512-325a6ee2-432c-4f65-8b9b-30ddf9001366.png)


I had to check for a `nullptr` if I tried to use a value I got from the `result` list.

I will look into the `GraphModelIntegrationTests, GetNodesFromGraphNodeIds` test when I get some time.

Reply:
Good catch! Yeah it looks like `GraphController::GetSelectedNodes` should probably be using `SceneRequests::GetSelectedNodes` instead of `SceneRequests::GetSelectedItems` but I think this is a good change as well.